### ISSUE TITLE
Support Jdk 1.7 in Spring module

### ISSUE THREAD

**Issue by kerraway** [database-rider/database-rider#220]
First of all, thanks for this fantastic jar.

I have tried database-rider in our projects that used jdk 1.8, that was great. 

And I want to use database-rider in some old projects that used jdk 1.7. So, can database-rider support jdk 1.7?

**Comment by rmpestano** [comment]
Hi @kerraway,

it should be possible to use jdk1.7, what issues are you having?

**Comment by kerraway** [comment]
When I use JDK 1.7, I get the 'major.minor version 52.0' error. I think it's because the database-rider jar was compiled in JDK 1.8.

```
objc[22440]: Class JavaLaunchHelper is implemented in both /Library/Java/JavaVirtualMachines/zulu-7.jdk/Contents/Home/bin/java (0x10e2b74c0) and /Library/Java/JavaVirtualMachines/zulu-7.jdk/Contents/Home/jre/lib/libinstrument.dylib (0x10e33f4d0). One of the two will be used. Which one is undefined.
Exception in thread "main" java.lang.UnsupportedClassVersionError: com/github/database/rider/spring/api/DBRider : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:808)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:443)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:65)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:349)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:348)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:430)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:329)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:363)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:278)
	at sun.reflect.generics.factory.CoreReflectionFactory.makeNamedType(CoreReflectionFactory.java:114)
	at sun.reflect.generics.visitor.Reifier.visitClassTypeSignature(Reifier.java:125)
	at sun.reflect.generics.tree.ClassTypeSignature.accept(ClassTypeSignature.java:49)
	at sun.reflect.annotation.AnnotationParser.parseSig(AnnotationParser.java:434)
	at sun.reflect.annotation.AnnotationParser.parseAnnotation2(AnnotationParser.java:239)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations2(AnnotationParser.java:117)
	at sun.reflect.annotation.AnnotationParser.parseAnnotations(AnnotationParser.java:70)
	at java.lang.Class.initAnnotationsIfNecessary(Class.java:3281)
	at java.lang.Class.getAnnotations(Class.java:3250)
	at org.junit.runner.Description.createSuiteDescription(Description.java:124)
	at org.junit.internal.runners.ErrorReportingRunner.getDescription(ErrorReportingRunner.java:28)
	at org.junit.runners.Suite.describeChild(Suite.java:123)
	at org.junit.runners.Suite.describeChild(Suite.java:27)
	at org.junit.runners.ParentRunner.getDescription(ParentRunner.java:352)
	at com.intellij.junit4.JUnit4IdeaTestRunner.getDescription(JUnit4IdeaTestRunner.java:78)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:50)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
```

**Comment by kerraway** [comment]
Not just the rider-spring, the rider-core-1.14.0.jar was compiled in JDK 11, its MANIFEST.MF:

```
Manifest-Version: 1.0
Archiver-Version: Plexus Archiver
Created-By: Apache Maven 3.6.2
Built-By: rpestano
Build-Jdk: 11.0.2
```

**Comment by rmpestano** [comment]
I think it was released with jdk 11 but in terms of source code it is [compatible with 1.7](https://github.com/database-rider/database-rider/blob/7f34ed3ceb649d830f000a0500fd7151c81a1d0d/pom.xml#L139-L143). I'll install jdk 1.7 and make some tests here. 

Thank you for the feedback

**Comment by rmpestano** [comment]
Hi again, for rider-core I could not reproduce, I just run [this project](https://github.com/database-rider/database-rider/tree/master/rider-examples/jpa-productivity-boosters) from rider-examples with jdk 7 and it just worked:

<img width="1174" alt="Screenshot 2020-07-07 at 11 53 50" src="https://user-images.githubusercontent.com/1592273/86764963-d1e75b00-c048-11ea-81f0-17681894a412.png">

Version `v1.15.2-SNAPSHOT` should now be compatible with jdk 1.7 for `rider-spring` as well. 

Can you try with `v1.15.2-SNAPSHOT` on a project using jdk 1.7 and rider-spring? you'll need to [enable snapshots on your build](https://github.com/database-rider/database-rider#snapshots) to use this version.

**Comment by kerraway** [comment]
> Hi again, for `rider-core` I could not reproduce, I just run [this project](https://github.com/database-rider/database-rider/tree/master/rider-examples/jpa-productivity-boosters) from rider-examples with jdk 7 and it just worked:
> 
> <img alt="Screenshot 2020-07-07 at 11 53 50" width="1174" src="https://user-images.githubusercontent.com/1592273/86764963-d1e75b00-c048-11ea-81f0-17681894a412.png">
> 
> Version `v1.15.2-SNAPSHOT` should now be compatible with jdk 1.7 for `rider-spring` as well.
> 
> Can you try with `v1.15.2-SNAPSHOT` on a project using jdk 1.7 and rider-spring? you'll need to [enable snapshots on your build](https://github.com/database-rider/database-rider#snapshots) to use this version.

Thanks a lot, I'll try it today.

**Comment by kerraway** [comment]
I run "mvn test" in jpa-productivity-boosters, test run successfully. Thanks for your work.

![image](https://user-images.githubusercontent.com/5738688/86887184-c3717000-c12a-11ea-95bb-2c8fd9e56974.png)

---

Another issue, I find errors in rider-core's test classes, such as com.github.database.rider.core.MsSQLDatabaseIt, com.github.database.rider.core.MySQLDatabaseIt, com.github.database.rider.core.MsSQLDatabaseIdentitySupportIt.

![image](https://user-images.githubusercontent.com/5738688/86887623-6b873900-c12b-11ea-864a-8aa2e19f25f1.png)

@rmpestano Do you have same errors? How can I solve them?

**Comment by rmpestano** [comment]
Thank you for the feedback, just released `v1.15.2` and it should be in maven central in next hours.
